File: src/main/java/toker/warbandscripts/panel/repository/PlayerRepository.java
package toker.warbandscripts.panel.repository;

import org.springframework.data.jpa.repository.JpaRepository;

import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.jpa.repository.Lock;
import org.springframework.data.jpa.repository.Query;
import toker.warbandscripts.panel.entity.Player;
import toker.warbandscripts.panel.entity.Server;

import javax.persistence.LockModeType;
import java.util.List;
import java.util.Optional;


public interface PlayerRepository extends BaseRepository<Player, Integer> {
    @Query("SELECT p FROM Player p " +
            "WHERE (" +
            "LOWER(p.name) LIKE CONCAT('%',LOWER(?1),'%') " +
            "OR CAST(p.uniqueId as text) LIKE CONCAT('%',?1,'%') " +
            "OR CAST(p.id as text) LIKE CONCAT('%',?1,'%')" +
            ")" +
            "AND p.faction.server = ?2")
    List<Player> likeSearch(String likeString, Integer server);

    @Override
    @Lock(LockModeType.OPTIMISTIC_FORCE_INCREMENT)
    <S extends Player> S saveAndFlush(S s);
}

File: src/main/java/toker/panel/repository/PlayerRepository.kt
package toker.panel.repository

import org.springframework.data.jpa.domain.Specification
import org.springframework.data.jpa.repository.Lock
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import toker.panel.entity.Player
import java.util.*
import javax.persistence.LockModeType
import javax.transaction.Transactional

interface PlayerRepository : BaseRepository<Player, Int> {
    @Query("SELECT p FROM Player p " +
            "WHERE (" +
            "LOWER(p.name) LIKE CONCAT('%',LOWER(?1),'%') " +
            "OR CAST(p.uniqueId as text) LIKE CONCAT('%',?1,'%') " +
            "OR CAST(p.id as text) LIKE CONCAT('%',?1,'%')" +
            ")" +
            "AND p.faction.server = ?2")
    fun likeSearch(likeString: String, server: Int): List<Player>

    @Query("""
        UPDATE Player p
        SET p.woundTime = current_timestamp, p.woundDuration = :duration
        WHERE p.id = :id
    """)
    @Modifying
    @Transactional
    fun wound(@Param("id") id: Int, @Param("duration") duration: Int)
}